text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+
+### Deprecation
+
+In theory, signals could be replaced. When this happens, they are marked as deprecated.
+
+Code is only marked as deprecated when the replacement becomes stable. Deprecated code still abides by the same support guarantees as stable code. Deprecated APIs remain stable and backwards compatible.
+
+### Removal
+
+Packages are end-of-life’d by being removed, along with a major version bump.
+
+We currently have no plans for deprecating signals or creating a major version past v1.0.
+
+## Version Numbers
+
+OpenTelemetry follows [semver 2.0](https://semver.org/) conventions, with the following distinction.
+
+For the purposes of versioning, all code within a release is treated as if it were part of a single package. All packages within a release receive the same version number. By keeping version numbers in lockstep, end users can understand which packages belong to which release.
+
+Note that contrib packages are released separately from core packages. Contrib releases are not required to match version numbers with the core releases.
+
+In some languages, package managers react poorly to experimental packages having a version higher than 0.X. In these cases, a language-specific workaround is required.
+
+Note: different languages implementations do not need to have matching version numbers, nor do implementations have to match the version of the specification they implement. For example, it is fine to have Python at 1.2.8, Java at 1.3.2, and the spec at 1.1.1. Within python, all packages in the 1.2.8 release have the version number 1.2.8, even if they are experimental.
+
+**Major version bump**  
+Major version bumps only occur when there is a breaking change to a stable interface, or the removal of deprecated signals.
+
+OpenTelemetry values long term support. The expectation is that we will version to v1.0 once the first set of packages are declared stable. OpenTelemetry will then remain at v1.0 for years. THere are no plans for a v2.0 of OpenTelemetry at this time. Additional stable packages, such as metrics and logs, will be added as minor version bumps.
+
+**Minor version bump**  
+Most changes to OpenTelemetry result in a minor version bump.
+
+* New backward-compatible functionality added to any component.
+* Breaking changes to internal SDK components.
+* Breaking changes to experimental signals.
+* New experimental packages are added.
+* Experimental packages become stable.
+
+**Patch version bump**  
+Patch versions make no changes which would require recompilation or potentially break application code. The following are examples of patch fixes. 
+
+* Bugs
+* Security fixes
+* Documentation
+
+Currently, OpenTelemetry does have plans to backport bug and security fixes to prior minor versions. Security and bug fixes are only applied to the latest minor version. We are committed to making it feasible for end users to stay up to date with the latest version of OpenTelemetry.
```

Comment:
In the hypothetical world in which we cut a new major version, and there was a security vulnerability, I think we would want to patch the latest minor version **in both major versions**.  It would be hard to force customers to upgrade to a new major version to get a security fix, as major version upgrades will be disruptive by definition.

Reply:
So, I agree that back-porting across major versions seems both reasonable and likely. But how many major versions? I would rather not pick something arbitrary, and leave it open for the future when we come to it. We can always add this clarification later.

Reply:
In my doc I had proposed all major versions that have had a minor release in the past year, but starts to get into the LTS discussion, which may be too much for this proposal.  It might be easier to start with something optional, like "it is up to maintainers to decide whether to support additional minor versions, or to support minor versions from previous major versions"